src/TSet.ts

```diff
@@ -103,6 +105,17 @@ export class TSet<T extends Eq<T>> implements Iterable<T> {
     return undefined;
   }
 
+  /**
+   * Returns the only T in the set. Returns undefined if the set is
+   * empty. Throws if the set has more than one member.
+   */
+  only(): Maybe<T> {
```

Comment:
I don't see the need for this method - it seems to be pushing non-TSet concerns into TSet and doesn't fit naturally with the general idea of a set that can have any number of elements. If clients of `TSet` expect there to be at most one element then client code should check the size and take remedial action directly, since it's the semantics of the client code that needs to treat multiple elements as an error, rather than the semantics of TSet. 

I recommend removing this method and handling it in client code.

Reply:
OK. I was modelling this off C#'s Enumerable.Single, which is a useful utility method even if it doesn't entirely fit within the semantics of a list, but it's not important.